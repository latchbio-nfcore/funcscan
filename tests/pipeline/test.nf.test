nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"
    tag "test"
    tag "pipeline"

    test("AMP and ARG workflow") {

        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(UTILS.removeNextflowVersion("$outputDir")).match("software_versions") },
                { assert snapshot(path("$outputDir/amp/ampir/").list()).match("amp_ampir") },
                { assert snapshot(path("$outputDir/amp/amplify/").list()).match("amp_amplify") },
                { assert new File("$outputDir/amp/hmmer_hmmsearch/sample_1/sample_1_mybacteriocin.txt.gz").exists() },
                { assert new File("$outputDir/amp/hmmer_hmmsearch/sample_2/sample_2_mybacteriocin.txt.gz").exists() },
                { assert snapshot(path("$outputDir/amp/macrel/sample_1/sample_1_log.txt"),
                                path("$outputDir/amp/macrel/sample_1/README.md")).match("amp_macrel_sample_1") },
                { assert new File("$outputDir/amp/macrel/sample_1/sample_1.all_orfs.faa.gz").exists() },
                { assert new File("$outputDir/amp/macrel/sample_1/sample_1.prediction.gz").exists() },
                { assert snapshot(path("$outputDir/amp/macrel/sample_2/sample_2_log.txt"),
                                path("$outputDir/amp/macrel/sample_2/README.md")).match("amp_macrel_sample_2") },
                { assert new File("$outputDir/amp/macrel/sample_2/sample_2.all_orfs.faa.gz").exists() },
                { assert new File("$outputDir/amp/macrel/sample_2/sample_2.prediction.gz").exists() },
                { assert snapshot(path("$outputDir/arg/abricate/").list()).match("arg_abricate") },
                { assert snapshot(path("$outputDir/arg/amrfinderplus/").list()).match("arg_amrfinderplus") },
                { assert snapshot(path("$outputDir/arg/fargene/sample_1/class_a/predictedGenes/wastewater_metagenome_contigs_1-class_A-filtered-peptides.fasta"),
                                path("$outputDir/arg/fargene/sample_1/class_a/predictedGenes/wastewater_metagenome_contigs_1-class_A-filtered.fasta"),
                                path("$outputDir/arg/fargene/sample_1/class_a/results_summary.txt"),
                                path("$outputDir/arg/fargene/sample_1/class_b_1_2/predictedGenes/wastewater_metagenome_contigs_1-class_B_1_2-filtered-peptides.fasta"),
                                path("$outputDir/arg/fargene/sample_1/class_b_1_2/predictedGenes/wastewater_metagenome_contigs_1-class_B_1_2-filtered.fasta"),
                                path("$outputDir/arg/fargene/sample_1/class_b_1_2/results_summary.txt"),
                                path("$outputDir/arg/fargene/sample_2/class_a/predictedGenes/wastewater_metagenome_contigs_2-class_A-filtered-peptides.fasta"),
                                path("$outputDir/arg/fargene/sample_2/class_a/predictedGenes/wastewater_metagenome_contigs_2-class_A-filtered.fasta"),
                                path("$outputDir/arg/fargene/sample_2/class_a/results_summary.txt"),
                                path("$outputDir/arg/fargene/sample_2/class_b_1_2/predictedGenes/wastewater_metagenome_contigs_2-class_B_1_2-filtered-peptides.fasta"),
                                path("$outputDir/arg/fargene/sample_2/class_b_1_2/predictedGenes/wastewater_metagenome_contigs_2-class_B_1_2-filtered.fasta"),
                                path("$outputDir/arg/fargene/sample_2/class_b_1_2/results_summary.txt")).match("arg_fargene") },
                { assert snapshot(path("$outputDir/arg/hamronization/").list()).match("arg_hamronization") },
                { assert snapshot(path("$outputDir/multiqc/multiqc_data/multiqc_citations.txt"),
                                path("$outputDir/multiqc/multiqc_data/multiqc_sources.txt")).match("multiqc") },
                { assert new File("$outputDir/reports/ampcombi/ampcombi.log").exists() },
                { assert new File("$outputDir/reports/ampcombi/sample_1/sample_1_amp.faa").exists() },
                { assert new File("$outputDir/reports/ampcombi/sample_1/sample_1_ampcombi.csv").exists() },
                { assert new File("$outputDir/reports/ampcombi/sample_1/sample_1_diamond_matches.txt").exists() },
                { assert new File("$outputDir/reports/ampcombi/sample_2/sample_2_amp.faa").exists() },
                { assert new File("$outputDir/reports/ampcombi/sample_2/sample_2_ampcombi.csv").exists() },
                { assert new File("$outputDir/reports/ampcombi/sample_2/sample_2_diamond_matches.txt").exists() },
                { assert snapshot(path("$outputDir/reports/hamronization_summarize/hamronization_combined_report.tsv")).match("summary_hamronization") }
            )
        }

    }

}
