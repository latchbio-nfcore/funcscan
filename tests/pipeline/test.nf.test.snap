{
    "amp_macrel_sample_2": {
        "content": [
            "sample_2_log.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
            "README.md:md5,fa3706dfc95d0538a52c4d0d824be5fb"
        ],
        "timestamp": "2023-11-17T09:45:44.507792541"
    },
    "arg_hamronization": {
        "content": [
            [
                "sample_1.txt.abricate.json:md5,6607ea5eafa36a435d955411228d6c9a",
                "sample_2.txt.abricate.json:md5,248ede666c7d658b2ff1b11c7aa05215"
            ],
            [
                "sample_1.tsv.amrfinderplus.json:md5,2276b9280cf4d976bbf3417f7f3cf6cd",
                "sample_2.tsv.amrfinderplus.json:md5,1abe9f30fd9856f66bab8afc01ac3640"
            ],
            [
                "sample_1_retrieved-genes-class_A-hmmsearched.out.fargene.json:md5,fa57ddf8c4d0e97311cfbc26c4cb32da",
                "sample_1_retrieved-genes-class_B_1_2-hmmsearched.out.fargene.json:md5,b22f390f828304a415b63142aba49294",
                "sample_1_wastewater_metagenome_contigs_1-class_A-hmmsearched.out.fargene.json:md5,8047786642b915f31a9c08cdbe2f4c63",
                "sample_1_wastewater_metagenome_contigs_1-class_B_1_2-hmmsearched.out.fargene.json:md5,910211226915539ded39f4d3b06b7b5c",
                "sample_2_retrieved-genes-class_A-hmmsearched.out.fargene.json:md5,3cbbca609957a85e4af8199c31efc996",
                "sample_2_retrieved-genes-class_B_1_2-hmmsearched.out.fargene.json:md5,e0a9f9a1ee104898caaa2fd587419c3d",
                "sample_2_wastewater_metagenome_contigs_2-class_A-hmmsearched.out.fargene.json:md5,b56cbf8a78ab7586259064731486f466",
                "sample_2_wastewater_metagenome_contigs_2-class_B_1_2-hmmsearched.out.fargene.json:md5,4563066087d2cc327e1fb95ec844de1a"
            ],
            [
                "sample_1.txt.rgi.json:md5,f41cdb118ab149bcce443bbde8badc48",
                "sample_2.txt.rgi.json:md5,604620d2d22b2b18f6208247daf6928c"
            ]
        ],
        "timestamp": "2023-11-17T09:45:44.599511516"
    },
    "amp_macrel_sample_1": {
        "content": [
            "sample_1_log.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
            "README.md:md5,fa3706dfc95d0538a52c4d0d824be5fb"
        ],
        "timestamp": "2023-11-17T09:45:44.489030004"
    },
    "software_versions": {
        "content": [
            "{ABRICATE_RUN={abricate=1.0.1}, AMPIR={ampir=1.1.0}, AMPLIFY_PREDICT={AMPlify=1.1.0}, AMP_HMMER_HMMSEARCH={hmmer=3.3.2}, AMRFINDERPLUS_RUN={amrfinderplus=3.11.18, amrfinderplus-database=2023-11-15.1}, AMRFINDERPLUS_UPDATE={amrfinderplus=3.11.18}, BIOAWK={bioawk=1.0}, CUSTOM_DUMPSOFTWAREVERSIONS={python=3.11.4, yaml=6.0}, FARGENE={fargene=0.1}, GUNZIP_FASTA_PREP={gunzip=1.10}, GUNZIP_MACREL_ORFS={gunzip=1.10}, GUNZIP_MACREL_PRED={gunzip=1.10}, HAMRONIZATION_ABRICATE={hamronization=1.1.1}, HAMRONIZATION_AMRFINDERPLUS={hamronization=1.1.1}, HAMRONIZATION_FARGENE={hamronization=1.1.1}, HAMRONIZATION_RGI={hamronization=1.1.1}, HAMRONIZATION_SUMMARIZE={hamronization=1.1.1}, MACREL_CONTIGS={macrel=1.2.0}, PRODIGAL_GFF={pigz=2.6, prodigal=2.6.3}, RGI_MAIN={rgi=5.2.1, rgi-database=3.1.4}, Workflow={nf-core/funcscan=1.1.5dev}}"
        ],
        "timestamp": "2023-11-17T12:26:24.217761567"
    },
    "amp_ampir": {
        "content": [
            [
                "sample_1.ampir.faa:md5,6ba608402fe88c6c630ea3404f6af3a4",
                "sample_1.ampir.tsv:md5,2b79534f61bd210b1df80320ca1a861e"
            ],
            [
                "sample_2.ampir.faa:md5,c7efd0d9b54a6c6acd56a66d65c4599d",
                "sample_2.ampir.tsv:md5,2dbbf8b83bba7703aefbdb43515c8896"
            ]
        ],
        "timestamp": "2023-11-17T09:45:44.433598026"
    },
    "arg_abricate": {
        "content": [
            [
                "sample_1.txt:md5,427cec26e354ac6b0ab6047ec6621202"
            ],
            [
                "sample_2.txt:md5,4c140c932a48a22bcd8ae911bda8f4c7"
            ]
        ],
        "timestamp": "2023-11-17T09:45:44.526839168"
    },
    "arg_amrfinderplus": {
        "content": [
            [
                "sample_1.tsv:md5,29cfb6f34f420d802eda95c6d9daa361"
            ],
            [
                "sample_2.tsv:md5,d9b6565167d603a1f07cff2374db8eb2"
            ]
        ],
        "timestamp": "2023-11-17T09:45:44.55438353"
    },
    "summary_hamronization": {
        "content": [
            "hamronization_combined_report.tsv:md5,0562e05ff0dcc9d241386503e507cd1a"
        ],
        "timestamp": "2023-11-17T09:45:44.685378463"
    },
    "multiqc": {
        "content": [
            "multiqc_citations.txt:md5,4c806e63a283ec1b7e78cdae3a923d4f",
            "multiqc_sources.txt:md5,d2a044df39ce3c6abe5cdc2d67473490"
        ],
        "timestamp": "2023-11-17T09:45:44.643748638"
    },
    "amp_amplify": {
        "content": [
            [
                "sample_1.tsv:md5,ceb5a6b25a41b4e46b1e1f3c8932f01b"
            ],
            [
                "sample_2.tsv:md5,071c6e2d2365d828c583fc14be5da724"
            ]
        ],
        "timestamp": "2023-11-17T09:45:44.466130494"
    },
    "arg_fargene": {
        "content": [
            "wastewater_metagenome_contigs_1-class_A-filtered-peptides.fasta:md5,0900b2ee4a2e1d5c36192532dd4a606b",
            "wastewater_metagenome_contigs_1-class_A-filtered.fasta:md5,9fe97d66263db17ab0233ca1203f1f5f",
            "results_summary.txt:md5,2c8a073d2a7938e8aedcc097e6df2aa5",
            "wastewater_metagenome_contigs_1-class_B_1_2-filtered-peptides.fasta:md5,20d374cae60e71b1195f7a2f1457f589",
            "wastewater_metagenome_contigs_1-class_B_1_2-filtered.fasta:md5,a3d620826a68ebb11fbfc428517b504a",
            "results_summary.txt:md5,59f2e69c670d72f0c0a401e0dc90cbeb",
            "wastewater_metagenome_contigs_2-class_A-filtered-peptides.fasta:md5,d4d937f9bbe507d2d644c70b6f24c487",
            "wastewater_metagenome_contigs_2-class_A-filtered.fasta:md5,aa29ac21de3d5d88339f7a06e24dea50",
            "results_summary.txt:md5,3b86a5513e89e22a4c8b9279678ce0c0",
            "wastewater_metagenome_contigs_2-class_B_1_2-filtered-peptides.fasta:md5,36e70d255bfd9d0ad46ac08b0ee4b0b6",
            "wastewater_metagenome_contigs_2-class_B_1_2-filtered.fasta:md5,8b1240dc4240a45e9e9dff6b12e24941",
            "results_summary.txt:md5,59f2e69c670d72f0c0a401e0dc90cbeb"
        ],
        "timestamp": "2023-11-17T09:45:44.575224104"
    }
}